engine/api/application_variable.go

```diff
@@ -144,34 +165,39 @@ func (api *API) updateVariableInApplicationHandler() service.Handler {
 			return sdk.WithStack(sdk.ErrWrongRequest)
 		}
 
-		app, err := application.LoadByName(api.mustDB(), key, appName)
+		proj, err := project.Load(api.mustDB(), projectKey)
 		if err != nil {
-			return sdk.WrapError(err, "Cannot load application: %s", appName)
+			return sdk.WrapError(err, "cannot load project %s", projectKey)
+		}
+
+		app, err := application.LoadByProjectIDAndName(ctx, api.mustDB(), proj.ID, appName)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load application: %s", appName)
 		}
 		if app.FromRepository != "" {
 			return sdk.WithStack(sdk.ErrForbidden)
 		}
 
-		variableBefore, err := application.LoadVariableWithDecryption(api.mustDB(), app.ID, newVar.ID, varName)
+		variableBefore, err := application.LoadVariableWithDecryption(ctx, api.mustDB(), app.ID, newVar.ID, varName)
```

Comment:
to load variable, we don't need to load the project, then application and finally the variable.
`projectKey`, `appName` and `varName` are identifiers

Reply:
Project is no more loaded